lib/Persistence/TrafficLimiter.php

```diff
@@ -83,6 +107,35 @@ public static function getHash($algo = 'sha512')
         return hash_hmac($algo, $_SERVER[self::$_ipKey], ServerSalt::get());
     }
 
+    /**
+     * Validate $_ipKey against configured ipranges. If matched ratelimiter will ignore ip
+     *
+     * @access private
+     * @static
+     * @param  string $ipRange
+     * @return bool
+     */
+    private static function matchIp($ipRange = null)
+    {
+        // Match $_ipKey to $ipRange and if it matches it will return with a true
+        $address = \IPLib\Factory::addressFromString($_SERVER[self::$_ipKey]);
+        $range   = \IPLib\Factory::rangeFromString(trim($ipRange));
+
+        // If $range is null something went wrong (possible invalid ip given in config). It's here becaue matches($range) does not accepts null vallue
+        if ($range == null) {
+            return false;
```

Comment:
Should not you/could not you log this error or somehow then let sysadmins know they should fix this?

(Thinking about PHP again and given we have no log file or so… I guess you cannot, so you can forget that.)

It’s a theoretical room for improvement though as silently ignoring errors is usually never nice…

Reply:
PHP errors can of course be (and usually are) logged, but that is up to the configuration of the webserver/php service.

In this case I agree not to error out.